### PR TITLE
Add implementation for ws_value_bool type

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #68

### CODE REVIEW

File: src/values/bool.c [hunk 1]
```diff
@@ -25,5 +25,65 @@
...
 
+/**
+ * Deinit callback for ws_value_bool type
+ */
+static void
+bool_deinit(
+    struct ws_value* self
+) {
+    if (self) {
+        if (self->type == WS_VALUE_TYPE_BOOL) {
```
Comment (neithernut) [inline src/values/bool.c]: You can savely assume that the type is, indeed, `WS_VALUE_TYPE_BOOL`.

File: src/values/bool.h [hunk 2]
```diff
@@ -34,7 +34,57 @@
...
+/**
+ * ws_value_bool type definition
+ */
+struct ws_value_bool {
+    struct ws_value value; //!< Supertype
+
+    struct {
+        bool b; //!< Actual value
+        pthread_rwlock_t rwl; //!< Read-Write lock for atomic operations
+    } boolean; //!< Thread-safe boolean type
```
Comment (neithernut) [inline src/values/bool.h]: Why the encaplulation?

Comment (matthiasbeyer) [inline src/values/bool.h]: For clearification what belongs to what.

Comment (neithernut) [inline src/values/bool.h]: It is not neccessary. We _always_ want to lock the entire type, so we always lock the value.

---

Should we put the lock into the `value`?

Comment (matthiasbeyer) [inline src/values/bool.h]: > Should we put the lock into the `value`?

I think so. Shall I open a new PR for this, or can it go in this one?

**Comment by neithernut** [comment]
I'm not 100% sure abot this yet, as objects are lockable on their own.
Maybe we really should have the locks in the basic datatype values. However, the data should not be bundeled in an anonymous struct.

File: src/values/bool.h (hunk 2 above)
Comment (neithernut) [inline src/values/bool.h]: Forget about this. I was wrong here, as I commented somewhere else.

---

The lock should still be removed, though.

Comment (matthiasbeyer) [inline src/values/bool.h]: So I guess I can remove the locking for the boolean (the actual `bool`), right?

Comment (neithernut) [inline src/values/bool.h]: Yep.

**Comment by matthiasbeyer** [comment]
I'm ready with this. ACK for squashing?

File: src/values/bool.c [hunk 3]
```diff
@@ -25,5 +25,58 @@
...
+#include <stdbool.h>
+
+#include "util/attributes.h"
 #include "values/bool.h"
 
+/**
+ * Deinit callback for ws_value_bool type
+ */
+static void
+bool_deinit(
```
Comment (neithernut) [inline src/values/bool.c]: Ame as for int. No `deinit` needed here.

**Comment by neithernut** [comment]
GO for squashing.

**Comment by matthiasbeyer** [comment]
Done.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**